code/framework/java-server/src/main/java/io/github/ibuildthecloud/gdapi/context/ApiContext.java

```diff
@@ -91,4 +97,19 @@ public TransformationService getTransformationService() {
     public void setTransformationService(TransformationService transformationService) {
         this.transformationService = transformationService;
     }
+
+    public void addCapability(Object object, String capability) {
+        if (dynamicCapabilities.get(object) == null){
```

Comment:
We have a dumb util method for this: `io.cattle.platform.util.type.CollectionUtils.addToMap(Map<K, V>, K, T, Class<? extends Collection>)`